### ISSUE TITLE
Something wrong after https://github.com/ericpaulbishop/gargoyle/commit/b332ebc7b25f495873cdb47c2927ba302712c042

### ISSUE THREAD

**Issue by obsy** [ericpaulbishop/gargoyle#916]
```
root@Gargoyle:~# gargoyle_header_footer -h -s "connection" -p "basic" -j "basic.js table.js" -z "basic.js" -i gargoyle network wireless dhcp firewall
Content-Type: text/html; charset=utf-8

<!DOCTYPE html>
<head>
	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge">
	<meta name="description" content="Gargoyle Firmware Webgui for router management.">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<title>Gargoyle Router Management Utility - Podstawowa</title>
	<link rel="shortcut icon" href="/themes/Gargoyle/images/favicon.png"/>
	<link rel="stylesheet" href="/themes/Gargoyle/bootstrap.min.css?1.13.0.0pre9--b332ebc7-">
	<link rel="stylesheet" href="/themes/Gargoyle/common.css?1.13.0.0pre9--b332ebc7-"/>
	<link rel="stylesheet" href="/themes/Gargoyle/theme.css?1.13.0.0pre9--b332ebc7-">
	<script src="/js/common.js?1.13.0.0pre9--b332ebc7-"></script>
	<script src="/js/basic.js?1.13.0.0pre9--b332ebc7-"></script>
	<script src="/js/table.js?1.13.0.0pre9--b332ebc7-"></script>
	<script src="/i18n/English-EN/strings.js?1.13.0.0pre9--b332ebc7-"></script>
	<script src="i18n/Polish-PL/strings.js?1.13.0.0pre9--b332ebc7-"></script>
	<script src="/i18n/English-EN/basic.js?1.13.0.0pre9--b332ebc7-"></script>
	<script src="/i18n/Polish-PL/basic.js?1.13.0.0pre9--b332ebc7-"></script>
	<script src="/themes/Gargoyle/theme.js?1.13.0.0pre9--b332ebc7-"></script>
</head>
<body>
	<div id="darken"><iframe id="d_iframe" class="select_free"></iframe></div>
	<div id="wait_msg">
		<div id="wait_txt">
			Proszę czekać na wprowadzenie zmian
		</div>
		<div id="wait_icon">
			<img src="/themes/Gargoyle/images/wait_icon.gif"/>
		</div>
		<iframe id="m_iframe" class="select_free"></iframe>
	</div>
	<div id="row-offcanvas" class="row-offcanvas full-height">
		<div id="wrapper" class="container-fluid full-height">
			<div id="content" class="col-xs-12 col-md-10 col-lg-10 col-md-push-2 col-lg-push-2 full-height">
				<div id="topnavbar" class="navbar navbar-default">
					<div class="container-fluid">
						<div class="navbar-header">
							<button type="button" class="btn btn-default sidebar-toggle navbar-toggle" onclick="sidebar()">
								<span class="sr-only">Toggle navigation</span>
								<span class="icon-bar"></span>
								<span class="icon-bar"></span>
								<span class="icon-bar"></span>
							</button>
							<span class="navbar-brand">Router Management Utility</span>
						</div>
					</div>
				</div>
				<div class="row">
					<div class="col-lg-12">
						<div class="alert alert-danger" style="display:none;">Pamięć flash routera jest zapełniona (zamontowania w trybie tylko do odczytu) lub tymczasowo wykorzystywana jest pamięć RAM. Router nie będzie działać poprawnie lub powróci do stanu poprzedniego po ponownym uruchomieniu. Przywróć domyślną konfigurację lub wgraj nowe oprogramowanie (bez zachowania ustawień).</div>
<script>
	var gargoyleBinRoot = "/www/.";
	var haveCollapsibleMenus = 1;
	var haveThemeJs = 1;
**Segmentation fault**
```

```
root@Gargoyle:~# cat /tmp/sysinfo/model 
GL-iNet 6416A v1
```

Wireless/network/firewall looks normal.

**Comment by lantis1008** [comment]
Can you provide "ubus -S call network.interface dump" please?

**Comment by lantis1008** [comment]
Also what mode are you in? DHCP wired/wireless? Repeater?

**Comment by obsy** [comment]
Default config, fresh after sysupgrade, without saving config, dhcp/wired.


`{"interface":[{"interface":"lan","up":true,"pending":false,"available":true,"autostart":true,"dynamic":false,"uptime":60,"l3_device":"br-lan","proto":"static","device":"br-lan","updated":["addresses"],"metric":0,"dns_metric":0,"delegation":true,"ipv4-address":[{"address":"192.168.1.1","mask":24}],"ipv6-address":[],"ipv6-prefix":[],"ipv6-prefix-assignment":[{"address":"fd3d:bd65:aa27::","mask":60,"local-address":{"address":"fd3d:bd65:aa27::1","mask":60}}],"route":[],"dns-server":[],"dns-search":[],"neighbors":[],"inactive":{"ipv4-address":[],"ipv6-address":[],"route":[],"dns-server":[],"dns-search":[],"neighbors":[]},"data":{}},{"interface":"loopback","up":true,"pending":false,"available":true,"autostart":true,"dynamic":false,"uptime":60,"l3_device":"lo","proto":"static","device":"lo","updated":["addresses"],"metric":0,"dns_metric":0,"delegation":true,"ipv4-address":[{"address":"127.0.0.1","mask":8}],"ipv6-address":[],"ipv6-prefix":[],"ipv6-prefix-assignment":[],"route":[],"dns-server":[],"dns-search":[],"neighbors":[],"inactive":{"ipv4-address":[],"ipv6-address":[],"route":[],"dns-server":[],"dns-search":[],"neighbors":[]},"data":{}},{"interface":"wan","up":false,"pending":false,"available":true,"autostart":true,"dynamic":false,"proto":"dhcp","device":"eth0","data":{}},{"interface":"wan6","up":false,"pending":false,"available":true,"autostart":true,"dynamic":false,"proto":"dhcpv6","device":"eth0","data":{}}]}`

**Comment by lantis1008** [comment]
Ok i think i understand this error. Will take a look after work and come back to you. Sorry for the inconvenience!

**Comment by lantis1008** [comment]
Can you please check if the following commit fixes the issue?
https://github.com/ericpaulbishop/gargoyle/commit/c59260f9f655920abf70b7402f1b622cc8d2b4c4

The json library states that some functions are NULL safe. Apparently I should have been more careful.

Note to self: maskbits and tmpobj2 can be dropped outside the new function as well.

**Comment by obsy** [comment]
In a few hours.

**Comment by obsy** [comment]
Ok, works.

**Comment by obsy** [comment]
And next issue:
```
Wed Mar 10 14:07:46 2021 daemon.info dnsmasq-dhcp[8930]: read /etc/ethers - 0 addresses
Wed Mar 10 14:07:47 2021 daemon.err uhttpd[3812]: /usr/lib/gargoyle/restart_network.sh: line 72: network_get_gateway: not found
Wed Mar 10 14:07:47 2021 daemon.err uhttpd[3812]: /usr/lib/gargoyle/restart_network.sh: line 73: network_get_gateway: not found
Wed Mar 10 14:07:48 2021 daemon.err uhttpd[3812]: sh: write error: Broken pipe

```

restart_network.sh  require . /lib/functions/network.sh

**Comment by lantis1008** [comment]
Fixed by 396836cd8dba90ba1f8ceafa51e4384738b5a126
